pkg/distributor/push_test.go

```diff
@@ -847,6 +846,123 @@ func TestHandler_ErrorTranslation(t *testing.T) {
 	}
 }
 
+func TestHandler_HandleRetryAfterHeader(t *testing.T) {
+	const (
+		defaultRetryBase      = 3 * time.Second
+		defaultMaxAllowedAtts = 2
+	)
+
+	testCases := []struct {
+		name          string
+		responseCode  int
+		retryAttempt  string
+		retryCfg      RetryConfig
+		expectRetry   bool
+		minRetryAfter int
+		maxRetryAfter int
+	}{
+		{
+			name:         "Request canceled, HTTP 499, no Retry-After",
+			responseCode: http.StatusRequestTimeout,
+			retryAttempt: "1",
+			retryCfg:     RetryConfig{Enabled: true, Base: defaultRetryBase, MaxAllowedAttempts: defaultMaxAllowedAtts},
```

Comment:
All tests use the same retryCfg, only Enabled field is different. Can we prepare two versions (`retryCfgEnabled`, `retryCfgDisabled`), and use those?

Reply:
We could also include some tests with different values in RetryConfig.

Reply:
I am changing retryAttempt in order to play with the senarios. I could keep the same test case just make it variable if it is prefered.